ui/src/components/table/Groupings-table.tsx

```diff
@@ -0,0 +1,141 @@
+'use client';
+import {
+    useReactTable,
+    flexRender,
+    getCoreRowModel,
+    getPaginationRowModel,
+    getFilteredRowModel,
+    getSortedRowModel,
+    SortingState
+} from '@tanstack/react-table';
+import { Table, TableBody, TableCell, TableHead, TableHeader, TableRow } from '@/components/ui/table';
+import ColumnSettings from '@/components/table/table-element/column-settings';
+import GroupingsTableHeaders from '@/components/table/table-element/groupings-table-headers';
+import PaginationBar from '@/components/table/table-element/pagination';
+import GlobalFilter from '@/components/table/table-element/global-filter';
+import SortArrow from '@/components/table/table-element/sort-arrow';
+import { useState } from 'react';
+import { SquarePen } from 'lucide-react';
+import GroupingPathCell from '@/components/table/table-element/grouping-path-cell';
+import { GroupingPath } from '@/models/groupings-api-results';
+import Link from 'next/link';
+
+const GroupingsTable = ({ groupingPaths }: { groupingPaths: GroupingPath[] }) => {
+    const [globalFilter, setGlobalFilter] = useState('');
+    const [sorting, setSorting] = useState<SortingState>([]);
+
+    const table = useReactTable({
+        columns: GroupingsTableHeaders,
+        data: groupingPaths,
+        getCoreRowModel: getCoreRowModel(),
+        getPaginationRowModel: getPaginationRowModel(),
+        getFilteredRowModel: getFilteredRowModel(),
+        getSortedRowModel: getSortedRowModel(),
+        state: { globalFilter, sorting },
+        initialState: { pagination: { pageSize: 20 } },
+        onGlobalFilterChange: setGlobalFilter,
+        onSortingChange: setSorting,
+        enableMultiSort: true
+    });
+
+    const columnCount = table.getHeaderGroups()[0].headers.length;
```

Comment:
You can use the `.length` of `GroupingsTableHeaders` instead here.

Reply:
done